xdem/coreg.py

```diff
@@ -1063,3 +1081,604 @@ def coregister(reference_raster: Union[str, gu.georaster.Raster], to_be_aligned_
+        if self.degree > 1:
+            raise ValueError(
+                "Nonlinear deramping degrees cannot be represented as transformation matrices."
+                f" (max 1, given: {self.degree})")
+        if self.degree == 1:
+            raise NotImplementedError("Vertical shift, rotation and horizontal scaling has to be implemented.")
+
+        # If degree==0, it's just a bias correction
+        empty_matrix = np.array([[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 1, 0], [0, 0, 0, 1]], dtype=float)
+
+        empty_matrix[2, 3] += self._meta["coefficients"][0]
+
+        return empty_matrix
+
+
+class CoregPipeline(Coreg):
+    """
+    A sequential set of coregistration steps.
+    """
+
+    def __init__(self, pipeline: list[Coreg]):  # pylint: disable=super-init-not-called
+        """
+        Instantiate a new coregistration pipeline.
+
+        :param: Coregistration steps to run in the sequence they are given.
+        """
+        self.pipeline = pipeline
+        self._meta = {}
+
+    def _fit_func(self, ref_dem: np.ndarray, tba_dem: np.ndarray, transform: Optional[rio.transform.Affine],
+                  weights: Optional[np.ndarray]):
+        """Fit each coregistration step with the previously transformed DEM."""
+        tba_dem_mod = tba_dem.copy()
+
+        for coreg in self.pipeline:
+            coreg._fit_func(ref_dem, tba_dem_mod, transform=transform, weights=weights)
+            coreg._fit_called = True
+
+            tba_dem_mod = coreg._apply_func(tba_dem_mod, transform)
+
+    def _apply_func(self, dem: np.ndarray, transform: rio.transform.Affine) -> np.ndarray:
+        """Apply the coregistration steps sequentially to a DEM."""
+        dem_mod = dem.copy()
+
+        for coreg in self.pipeline:
+            dem_mod = coreg.apply(dem_mod, transform)
+
+        return dem_mod
+
+    def _apply_pts_func(self, coords: np.ndarray) -> np.ndarray:
+        """Apply the coregistration steps sequentially to a set of points."""
+        coords_mod = coords.copy()
+
+        for coreg in self.pipeline:
+            coords_mod = coreg._apply_pts_func(coords_mod)
+
+        return coords_mod
+
+    def _to_matrix_func(self) -> np.ndarray:
+        """Try to join the coregistration steps to a single transformation matrix."""
+        if not _HAS_P3D:
+            raise ValueError("Optional dependency needed. Install 'pytransform3d'")
+
+        transform_mgr = TransformManager()
+
+        with warnings.catch_warnings():
+            # Deprecation warning from pytransform3d. Let's hope that is fixed in the near future.
+            warnings.filterwarnings("ignore", message="`np.float` is a deprecated alias for the builtin `float`")
+            for i, coreg in enumerate(self.pipeline):
+                new_matrix = coreg.to_matrix()
+
+                transform_mgr.add_transform(i, i + 1, new_matrix)
+
+            return transform_mgr.get_transform(0, len(self.pipeline))
```

Comment:
I am a bit afraid of matrices haha, so that's why I decided on using pytransform3d. Indeed, the syntax becomes a bit convoluted here, but I think (keywork think, I am not certain) it is necessary. Basically what we want is to merge transforms "A -> B -> C" into just "A-C" (in my case, I've used integer indices instead: "0 -> 1 -> 2" into "0-2"). 

Maybe there's an easier way to do this, but pytransform3d makes sure it is done right.